synapse/lib/cell.py

```diff
@@ -989,6 +994,24 @@ async def fini():
     async def initServiceRuntime(self):
         pass
 
+    async def procTask(self, func, *args, **kwargs):
+        '''
+        Execute a function in a process pool.
+
+        Args:
+            func: Function to execute. This must be able to be pickled.
+            *args: Arguments to the function.
+            **kwargs: Keyword arguments to the function.
+
+        Returns:
+            The function result.
+        '''
+        real = functools.partial(func, *args, **kwargs)
+        loop = asyncio.get_event_loop()
+        fut = loop.run_in_executor(self._procPool, real)
```

Comment:
curious if we can get / submit this from a executor thread

Reply:
why incur the extra context switch?

Reply:
The first call to the process pool does incur the overhead of spinning up the processes. Subseqent calls do end up doing queue submission. There is a unclear comment in the concurrent.future.ProcessPool code about the potential for the submission queue to block; and blocking in the ioloop is not good.